### PR TITLE
UAR-1425 Add Trusts Ceased Date and validation

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1425

### Change description
Add Feature flag for a new ceased date on trust-details page when reviewing trust info during an update.
If there are no eligible BOs (ie no BO checkboxes on the page) then we show an option to add a ceased date, but only if the feature flag is enabled - FEATURE_FLAG_ENABLE_CEASE_TRUSTS.

Also made a change to force the trust-details page to be re-shown if the user navigates back to an earlier point in the journey to change something. Previously it would not re-show the trust-details if it had already been submitted for that trust in review. This new feature is also enabled with the same feature flag.

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/update/update.manage.trusts.tell.us.about.the.legal.entity.controller.spec.ts [hunk 1]
```diff
@@ -142,7 +142,6 @@ mockGetApplicationData.mockReturnValue({
           reviewed_former_bos: false,
           reviewed_individuals: false,
           reviewed_legal_entities: false,
-          reviewed_trust_details: false,
```
Comment (markpit) [inline test/controllers/update/update.manage.trusts.tell.us.about.the.legal.entity.controller.spec.ts:145]: this was just a duplicate field that I noticed in the test object (if you expand the code window you can see it was in there twice)

File: src/utils/update/review_trusts.ts [hunk 2]
```diff
@@ -138,3 +138,13 @@ export const putTrustInChangeScenario = (appData: ApplicationData, trustId: stri
     };
   }
 };
+
+export const resetTrustInReviewPagesReviewed = (trustInReview: Trust) => {
```
Comment (mwestacott) [inline src/utils/update/review_trusts.ts:142]: Function name is not that clear to me. Would 'setAllTrustsAsNotReviewed' accurately describe what this is doing and is it any better?

Comment (markpit) [reply src/utils/update/review_trusts.ts:142]: it's only for the trust that is currently in review so not applicable to all trusts. Is resetPagesReviewedOnTrustInReview any better?

Comment (mwestacott) [reply src/utils/update/review_trusts.ts:142]: `resetPagesReviewed` maybe?

**Review by dlloyd1-ch — APPROVED** [review]
LGTM

**Comment by markpit** [comment]
i've missed a validation rule that wasn't in the AC's but in the comments of the jira story, cease date should not be before start date so working on that ATM

**Comment by markpit** [comment]
The extra ceased date cannot be before created date rule and tests has now been added in the latest commits (not the merge from main)

**Review by mwestacott — APPROVED** [review]

File: src/validation/error.messages.ts [hunk 3]
```diff
@@ -96,6 +96,12 @@ export enum ErrorMessages {
   YEAR_LENGTH_OF_TRUST = "The year of the date the trust was created must include 4 numbers",
   DAY_LENGTH_OF_TRUST = "The day the trust was created must include 1 or 2 numbers",
   MONTH_LENGTH_OF_TRUST = "The month the trust was created must include 1 or 2 numbers",
+  DAY_OF_TRUST_CEASED = "The date the trust stopped being associated with this overseas entity must include a day",
```
Comment (mwestacott) [inline src/validation/error.messages.ts:99]: Minor, but the reading of these constant names now sounds slightly off to me.

E.g. `DAY_OF_CEASED_TRUST` sounds preferable here. What do you think?

**Review by dlloyd1-ch — APPROVED** [review]

**Review by mwestacott — APPROVED** [review]

**Review by dlloyd1-ch — APPROVED** [review]

**Review by mwestacott — APPROVED** [review]

**Pull request merged**